### PR TITLE
New option to disable GraphQL schema token limit validation

### PR DESCRIPTION
**Pull request by yholkamp**
---

### Description

Related to #1265: This PR adds a way to configure the maximum number of GraphQL grammar tokens the parser will process before throwing a 'denial of service' prevention warning. While Gradle users can easily override this, with Maven this does not appear to be feasible.

---

Changes were made to:
- [ ] Codegen library - Java
- [ ] Codegen library - Kotlin
- [ ] Codegen library - Scala
- [x] Maven plugin
- [x] Gradle plugin
- [x] SBT plugin

### CODE REVIEW

**Comment by kobylynskyi** [comment]
@yholkamp thanks for creating a PR. Can you please add the same option to Gradle and SBT plugins? Thank you!

**Comment by yholkamp** [comment]
Apologies for the multiple pushes, the linter didn't quite agree with me.

@kobylynskyi thanks for your reply, after closer inspection of the GraphQL library, my initial approach seems to be too much of a shortcut. There are [4 distinct limits that can be configured on the parser](https://github.com/graphql-java/graphql-java/blob/86376aaee8fd507edd4e28e7a2a47b10b33b4e7a/src/main/java/graphql/parser/ParserOptions.java#L15-L53) and these require different values to be useful. The library imposes these to avoid scenarios where its used to process GraphQL queries on a server, which is very different from the purpose here. 

In order to make the plugin easy to use for the typical user, I've adjusted my changes to add a setting that skips this validation, while allowing it to be enabled when that's considered to be relevant. I'm hoping you'll agree with that approach.

File: plugins/gradle/graphql-java-codegen-gradle-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/gradle/GraphQLCodegenGradleTask.java [hunk 1]
```diff
@@ -121,6 +122,7 @@ public class GraphQLCodegenGradleTask extends DefaultTask implements GraphQLCode
     private String unknownFieldsPropertyName = MappingConfigConstants.DEFAULT_UNKNOWN_FIELDS_PROPERTY_NAME;
 
     private Boolean skip = false;
+    private Boolean skipSchemaSizeLimit = true;
```
Comment (kobylynskyi) [inline plugins/gradle/graphql-java-codegen-gradle-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/gradle/GraphQLCodegenGradleTask.java:125]: Please use `MappingConfigConstants.DEFAULT_` for the default value of this parameter (for consistency purposes).

File: plugins/maven/graphql-java-codegen-maven-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/GraphQLCodegenMojo.java [hunk 2]
```diff
@@ -248,6 +249,9 @@ public class GraphQLCodegenMojo extends AbstractMojo implements GraphQLCodegenCo
     @Parameter(defaultValue = "false")
     private boolean skip;
 
+    @Parameter(defaultValue = "true")
```
Comment (kobylynskyi) [inline plugins/maven/graphql-java-codegen-maven-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/GraphQLCodegenMojo.java:252]: Please use `MappingConfigConstants.DEFAULT_` for the default value of this parameter (for consistency purposes).

File: plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenPlugin.scala [hunk 3]
```diff
@@ -131,7 +132,8 @@ class GraphQLCodegenPlugin(configuration: Configuration, private[codegen] val co
     unknownFieldsPropertyName      := MappingConfigConstants.DEFAULT_UNKNOWN_FIELDS_PROPERTY_NAME,
     generateNoArgsConstructorOnly  := MappingConfigConstants.DEFAULT_GENERATE_NOARGS_CONSTRUCTOR_ONLY,
     generateModelsWithPublicFields := MappingConfigConstants.DEFAULT_GENERATE_MODELS_WITH_PUBLIC_FIELDS,
-    skip                           := false
+    skip                           := false,
+    skipSchemaSizeLimit            := true
```
Comment (kobylynskyi) [inline plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenPlugin.scala:136]: Please use `MappingConfigConstants.DEFAULT_` for the default value of this parameter (for consistency purposes).

File: plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenPlugin.scala [hunk 4]
```diff
@@ -247,6 +249,16 @@ class GraphQLCodegenPlugin(configuration: Configuration, private[codegen] val co
       graphqlCodegen := {
         sLog.value.info(s"Generating files: ${BuildInfo.toString}")
         val mappingConfigSupplier = buildJsonSupplier(configurationFiles.value)
+
+        if (skipSchemaSizeLimit.value) {
+          val parserOptionBuilder = ParserOptions
+            .newParserOptions()
+            .maxTokens(Integer.MAX_VALUE)
+            // .maxCharacters(Integer.MAX_VALUE)
```
Comment (kobylynskyi) [inline plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenPlugin.scala:257]: Was this intentional?

**Review by kobylynskyi — CHANGES_REQUESTED** [review]
@yholkamp thanks for the changes! Agree with your approach to disable the validation completely as too many users of this plugin really want this feature to be disabled.

**Review by kobylynskyi — APPROVED** [review]

**Comment by kobylynskyi** [comment]
@yholkamp thanks for working on this! Will be included in 5.9.0 release

**Pull request merged**